fdbserver/Status.actor.cpp

```diff
@@ -3669,6 +3670,45 @@ ACTOR Future<StatusReply> clusterGetStatus(
 		    .detail("Duration", timer() - tStart)
 		    .detail("StatusSize", statusObj.getFinalLength());
 
+		std::cout << "StatusObj string: " << statusObj.getJson() << std::endl;
+
+		return StatusReply(statusObj.getJson());
+
+	} catch (Error& e) {
+		TraceEvent(SevError, "StatusError").error(e);
+		throw;
+	}
+}
+
+StatusReply clusterGetFaultToleranceStatus(const std::string& statusStr) {
+	double tStart = timer();
+
+	try {
+		json_spirit::mValue mv = readJSONStrictly(statusStr);
+		JSONDoc jsonDoc(mv);
+
+		std::string faultToleranceRelatedFields[] = {
+			"fault_tolerance", "data",    "logs", "maintenance_zone", "maintenance_seconds_remaining",
+			"recovery_state",  "messages"
+		};
+
+		JsonBuilderObject statusObj;
+		for (std::string& field : faultToleranceRelatedFields) {
+			if (jsonDoc.has(field, false)) {
```

Comment:
is this right? The documentation of `has` says the parameter is the full path of the field.

Reply:
Yes, that's what the documentation says. I checked the code of "has()" and I think it will work even if it is not a complete path as long as this field is unique within the JSON document (here is the check it is using to locate the field: https://github.com/apple/foundationdb/blob/7295e1b806462396f0338c3c22e5f18931c5d78f/fdbclient/include/fdbclient/JSONDoc.h#L125). I can change this to be a complete path if we want to be safer about this.

Reply:
It doesn't look right if only the "key" is used without the full path. Can you change to use the complete path? There are cases the same key is used in different paths.

Reply:
Sure.

Also, doing "setKey()" below doesn't seem to work well with how StatusClient is updating the status (https://github.com/apple/foundationdb/blob/912707f4f22932287e5ae574954e2949d3848e53/fdbclient/StatusClient.actor.cpp#L552).The code in JsonDoc/json_spirit is throwing an exception (and the error text is not very helpful in identifying the problem), looking into this issue.

Reply:
> It doesn't look right if only the "key" is used without the full path. Can you change to use the complete path?

It turns out these fields are all top-level fields in the status json object (composed by this actor). They all come under "cluster" field in the status json object output by "fdbcli", that's because "cluster" field is added by the StatusClient. So I think we can let these fields be treated as complete paths. I will drop the "false" argument here.

Reply:
> Also, doing "setKey()" below doesn't seem to work well with how StatusClient is updating the status. The code in JsonDoc/json_spirit is throwing an exception (and the error text is not very helpful in identifying the problem), looking into this issue.

I think I figured out why. We are writing the object as a string, so StatusClient is not able to look into fields of that object (that is, not able to convert it into "json_spirit::mObject::Map").